src/transformers/pipelines.py

```diff
@@ -1112,10 +1126,17 @@ def group_entities(self, entities: List[dict]) -> List[dict]:
 
             # If the current entity is similar and adjacent to the previous entity, append it to the disaggregated entity group
             # The split is meant to account for the "B" and "I" suffixes
+            # Shouldn't merge if both entities are B-type
             if (
                 entity["entity"].split("-")[-1] == entity_group_disagg[-1]["entity"].split("-")[-1]
+                and entity["entity"].split("-")[0] != "B"
```

Comment:
```suggestion
```

Reply:
Now I'm thinking maybe this should be optional?

Reply:
@cceyda Actually, I reviewed again, and agree we should keep this here.